### ISSUE TITLE
stack smashing detected when running ner training

### ISSUE THREAD

**Issue by rohithkodali** [explosion/spaCy#2261]
I have spacy setup for noth python 2.7 and 3.5 but when i try to run ner training on spacy it throw the following error

Created blank 'en' model
Warning: Unnamed vectors -- this won't allow multiple vectors models to be loaded. (Shape: (0, 0))
*** stack smashing detected ***: python terminated
Aborted (core dumped)

where mught be it went wrong?

 

## Your Environment
* **Python version:** 2.7.12
* **Platform:** Linux-4.4.0-121-generic-x86_64-with-Ubuntu-16.04-xenial
* **spaCy version:** 2.0.12.dev0
* **Models:** en, te

**Comment by rohithkodali** [comment]
update:
 if we use only part of dataset  2300 sentences (in my case) it works without above error.

**Comment by rohithkodali** [comment]
hi @honnibal any update on this we still face the same issue when we used the more than 2300 sentences to train an NER

**Comment by kororo** [comment]
Hi @rohithkodali and @honnibal,

I had same issue, add the vector name before calling the begin_training.

```python
nlp.vocab.vectors.name = 'spacy_pretrained_vectors'
optimizer = nlp.begin_training()
```

The reason after investigating the source code, it gives you warning if you don't put name into vectors before calling begin_training.

https://github.com/explosion/spaCy/blob/61ef0739b8cede7bf87076f41fa1a2395ff52da2/spacy/_ml.py#L244-L250

Hope helps

**Comment by rohithkodali** [comment]
No luck still getting the same error, @kororo  did it solved for you?

**Comment by kororo** [comment]
@rohithkodali ah sorry, the snippet I gave, just for the warning fix that I found.

The 2300 sentences is related to something else. I am curious so I trained 2880 sentences and I have no problem at all.

- How many iteration and drop rate you have?
- Are you using GPU to train?

That error messages mostly happen on GPU computation. Try to check on your CUDA version. Otherwise, post some of your training code, probs I could spot something up.

**Comment by rohithkodali** [comment]
it stops at first iteration only when i use full daya but when i use 2300 sentenecs it worked well for iterations.
i use GPU to train.

`#!/usr/bin/env python
# coding: utf8
from __future__ import unicode_literals, print_function
import plac
import random
from pathlib import Path
import time

import spacy
import thinc.neural.gpu_ops
from nerDATA import *


def main(model='te', output_dir='telugu-nermodel6000v4', n_iter=50):
    """Load the model, set up the pipeline and train the entity recognizer."""
    if model is not None:
        nlp = spacy.load(model)  # load existing spaCy model
        print("Loaded model '%s'" % model)
    else:
        nlp = spacy.blank('te')  # create blank Language class
        print("Created blank 'te' model")

    # create the built-in pipeline components and add them to the pipeline
    # nlp.create_pipe works for built-ins that are registered with spaCy
    if 'ner' not in nlp.pipe_names:
        ner = nlp.create_pipe('ner')
        nlp.add_pipe(ner, last=True)
    # otherwise, get it so we can add labels
    else:
        ner = nlp.get_pipe('ner')

    # add labels
    for _, annotations in TRAIN_DATA:
        for ent in annotations.get('entities'):
            ner.add_label(ent[2])
    print(ner)


    # get names of other pipes to disable them during training
    other_pipes = [pipe for pipe in nlp.pipe_names if pipe != 'ner']
    with nlp.disable_pipes(*other_pipes):  # only train NER
        nlp.vocab.vectors.name = 'spacy_pretrained_vectors'
        optimizer = nlp.begin_training(device=0)
        for itn in range(n_iter):
            random.shuffle(TRAIN_DATA)
            losses = {}
            start_time=time.time()
            for text, annotations in TRAIN_DATA:
                print(text)
                nlp.update(
                    [text],  # batch of texts
                    [annotations],  # batch of annotations
                    drop=0.5,  # dropout - make it harder to memorise data
                    sgd=optimizer,  # callable to update weights
                    losses=losses)
            print(losses)
        print("time for iteration: ", time.time()-start_time)



    # save model to output directory
    if output_dir is not None:
        output_dir = Path(output_dir)
        if not output_dir.exists():
            output_dir.mkdir()
        nlp.to_disk(output_dir)
        print("Saved model to", output_dir)



if __name__ == '__main__':
    plac.call(main)
`


this is my code to train

**Comment by honnibal** [comment]
Thanks for the report. This will be difficult to track down, but I hope we can get to the bottom of it.

**Comment by kororo** [comment]
@rohithkodali I am more convince that something to do with your CUDA version since you are using GPU to train.

1. What is your CUDA version by calling "nvcc --version"
2. Could you remove your device=0 in "optimizer = nlp.begin_training(device=0)" and try whether this is confirm GPU?
3. What is your GPU spec? Is it in AWS?

From what I know, the latest CUDA version causes this issue.

Thanks

**Comment by rohithkodali** [comment]
@kororo  
1. here is my "nvcc --version" output

nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2016 NVIDIA Corporation
Built on Wed_May__4_21:01:56_CDT_2016
Cuda compilation tools, release 8.0, V8.0.26

2. i tried to use only CPU but still got the same error

3. my GPU is gtx 1080 on my local server (not using any cloud)

**Comment by ptamas88** [comment]
hello,
i have the same issue
@rohithkodali have you found a solution?
unfortunately, i have other problems with training existing model, so i cant feed the data in batches :(
thanks

**Comment by honnibal** [comment]
I believe this issue is now fixed :tada: . See here: https://github.com/explosion/spaCy/issues/2800

Thanks for your help reporting.
